Assets/Scripts/Character/PlayerHealth.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+
+public class PlayerHealthSystem : MonoBehaviour
+{
+    [Header("Health Settings")]
+    public float maxHealth = 100.0f;
+    public float currentHealth;
+    public float damagePerSecond = 10f;
+
+    private float _timeSinceLastDamage;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = maxHealth;
+        _timeSinceLastDamage = 0f;
+    }
+
+    public void TakeDamage(float damage)
+    {
+        currentHealth -= damage;
+
+        if (currentHealth <= 0)
+        {
+            Die();
+        }
+    }
+
+    private void Die()
+    {
+        // Implement any logic needed when the object dies, e.g., destroying the object
+        Destroy(gameObject);
+    }
+
+    void OnTriggerStay(Collider other)
+    {
```

Comment:
I get what you’re doing here but your naming is a bit confusing. Damage over time should be a different class based on the context. You’re trying to do invincibility frames here but it’s not clear when reading at a glance. This code shouldn’t be in OnTrigger, but should be in TakeDamage. You would just return before sending damage to the health bar with logic you have here minus the CompareTag.